client/src/components/NavBar/NavBar.tsx

```diff
@@ -4,27 +4,73 @@ import useStyles from './useStyles';
 import logo from '../../Images/logo.png';
 import { useAuth } from '../../context/useAuthContext';
 import { useSocket } from '../../context/useSocketContext';
-import { useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useEffect, useRef } from 'react';
 import LoggedInBar from './AuthBars/LoggedInBar';
 import LoggedOutBar from './AuthBars/LoggedOutBar';
 import CssBaseline from '@material-ui/core/CssBaseline';
+import HomePageBar from './AuthBars/HomePageBar';
+import { useHistory, useLocation } from 'react-router-dom';
 
 const NavBar = (): JSX.Element => {
   const classes = useStyles();
-
+  const history = useHistory();
+  const location = useLocation();
   const { loggedInUser } = useAuth();
   const { initSocket } = useSocket();
+  const outterNavStyle = useRef(classes.appbarHomePage);
+  const innerNavStyle = useRef(classes.appbarHomePage);
+
+  const history = useHistory();
+  const location = useLocation();
+
+  useEffect(() => {
+    const locationChange = history.listen((location) => {
```

Comment:
Some of this code if from other PRs, please update this branch with the current main branch :)